### PR TITLE
Navaid bugfixes and user fixes (POIs/VRPs)

### PR DESCRIPTION
**Pull request by dave6502**
Hi Folke, 
I've removed Obstacles import. Added manual load of a CSV file. Last CSV file loaded is saved in preferences and autoloaded on startup. The test/demo files sent to you before are still valid. I'm just writing up a short .md file which I'll email separately.
Thanks

### CODE REVIEW

File: src/libxdata/parsers/BaseParser.cpp [hunk 1]
```diff
@@ -42,10 +42,10 @@ std::string BaseParser::parseHeader() {
     getLine(line);
 
     std::istringstream lineStr(line);
-    int version;
+    char origin;
```
Comment (fpw) [inline src/libxdata/parsers/BaseParser.cpp:45]: Not sure if I understand this change - isn't origin the A or I that's parsed above? And the version only on a single line? Source for example here: https://developer.x-plane.com/article/flightplan-files-v11-fms-file-format/

File: src/libxdata/parsers/UserFixParser.cpp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    LOG_INFO(0, "type=%d, name='%s', ident='%s', %f, %f",
+            userFix.type, userFix.name.c_str(), userFix.ident.c_str(),
+            userFix.latitude, userFix.longitude);
+
+    acceptor(userFix);
+    lineNum++;
+}
+
+UserFix::Type UserFixParser::parseType(std::string& typeString) {
+    if ((typeString.compare("VRP") == 0) || (typeString.compare("11") == 0)) {
```
Comment (fpw) [inline src/libxdata/parsers/UserFixParser.cpp:92]: You could simply use the `==` operator on C++ strings, it's an overloaded operator with the expected meaning, even if the RHS is a string literal or C string.

File: src/maps/OverlayedUserFix.cpp [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    OverlayedFix(helper, fix){
+    if (POIIcon.getHeight() == 0) {
+        createPOIVRPIcons();
+    }
+}
+
+std::shared_ptr<OverlayedUserFix> OverlayedUserFix::getInstanceIfVisible(OverlayHelper helper, const xdata::Fix &fix) {
+    auto userFix = fix.getUserFix();
+    if (!userFix) {
+        return NULL;
```
Comment (fpw) [inline src/maps/OverlayedUserFix.cpp:37]: Please use `nullptr` instead.

**Review by fpw — COMMENTED** [review]
Fits perfectly into the architecture! Just a few minor things regarding modern C++ stuff.

**Comment by dave6502** [comment]
Made requested fixes for most comments and an explanation for the origin/version parsing

**Review by fpw — APPROVED** [review]
Thanks!

**Review by fpw — APPROVED** [review]

**Review by fpw — APPROVED** [review]
Looks all good to me, will merge it.

**Pull request merged**